spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/query/NearQuery.java

```diff
@@ -420,25 +426,54 @@ public DBObject toDBObject() {
 		}
 
 		if (maxDistance != null) {
-			dbObject.put("maxDistance", maxDistance.getNormalizedValue());
+			dbObject.put("maxDistance", getDistanceValueInRadiantsOrMeters(maxDistance));
 		}
 
 		if (minDistance != null) {
-			dbObject.put("minDistance", minDistance.getNormalizedValue());
+			dbObject.put("minDistance", getDistanceValueInRadiantsOrMeters(minDistance));
 		}
 
 		if (metric != null) {
-			dbObject.put("distanceMultiplier", metric.getMultiplier());
+			dbObject.put(
+					"distanceMultiplier",
+					usesMetricSystem() ? BigDecimal.ONE.divide(getMetricSystemNormalizer(metric), PRECISION + 3,
+							RoundingMode.HALF_UP).divide(new BigDecimal(1000), PRECISION + 3,
+							RoundingMode.HALF_UP).doubleValue() : metric.getMultiplier());
 		}
 
 		if (num != null) {
 			dbObject.put("num", num);
 		}
 
-		dbObject.put("near", Arrays.asList(point.getX(), point.getY()));
+		if (point instanceof GeoJsonPoint) {
```

Comment:
Still for precise calculations we need to rely on `BigDeciamal` no matter what `Distance` uses currently.